substrate/client/consensus/babe/src/lib.rs

```diff
@@ -1578,6 +1578,16 @@ where
 					viable_epoch.as_ref().start_slot,
 				);
 
+				if viable_epoch.as_ref().duration != self.config.epoch_length {
```

Comment:
I have to reason a bit how this behaves in case of skipped epochs (line 1547 above). As the logic there uses the duration in the `viable_epoch` for some internal stuff. Back to you soon

Reply:
yes. I moved the codes up